src/utils/test.ts

```diff
@@ -49,3 +62,139 @@ export function withTestScheduler(
     }),
   );
 }
+
+export function mockMember(member: Partial<RoomMember>): RoomMember {
+  return {
+    on() {
+      return this;
+    },
+    off() {
+      return this;
+    },
+    addListener() {
+      return this;
+    },
+    removeListener() {
+      return this;
+    },
+    ...member,
+  } as RoomMember;
+}
+
+export function mockMatrixRoom(room: Partial<MatrixRoom>): MatrixRoom {
```

Comment:
Can we have a generic `mockEventEmitter<T>(val: Partial<T>): T{}` and simplify those to:
```
export function mockMatrixRoom(room: Partial<MatrixRoom>): MatrixRoom {
    return mockEventEmitter(room);
}
```
(of course this is very much in conflict that this should be done in the js-sdk ... but still would shorten this a bit.)

Reply:
This is non blocking. (To merge this i think it is enough that the comment that we consider using js-sdk code here can also have a short sentence about a generic event emitter mock)